src/main/menu.js

```diff
@@ -364,5 +377,27 @@ export const setupMenu = store => {
     return item;
   });
 
-  Menu.setApplicationMenu(Menu.buildFromTemplate(template));
+  const buildTemplate = Menu.buildFromTemplate(template);
+
+  if (getSetting(state, 'isTrayModeEnabled')) {
+    if (!tray) {
+      tray = new Tray(trayIcon);
+    }
+
+    app.dock.hide();
+
+    tray.setContextMenu(buildTemplate);
+  } else {
+    app.dock.show();
+
+    setTimeout(() => {
+      if (app.dock.isVisible()) {
+        if (tray) {
+          tray.destroy();
+        }
+        tray = null;
+      }
+    }, 300);
```

Comment:
What's with the timeout? 😊

Reply:
app.dock.show() is asynchronous. When I disable the tray icon and enable the dock icon at the same time, the app crashes.

Reply:
Can you use `0` as the value? Having arbitrary values (300) is not so stable and the necessary value could differ from computer to computer.